datalad/distribution/tests/test_publish.py

```diff
@@ -713,3 +716,18 @@ def test_gh1811(srcpath, clonepath):
         path=clone.path, type='dataset', action='publish',
         status='impossible',
         message=('Cannot determine remote branch name from %s', 'HEAD'))
+
+
+@with_tempfile(mkdir=True)
+def test_publish_no_fetch_refspec_configured(path):
+    from datalad.cmd import GitRunner
+
+    path = Path(path)
+    GitRunner(cwd=str(path)).run(["git", "init", "--bare", "empty-remote"])
+    ds = Dataset(path / "ds").create()
+    ds.repo.add_remote("origin", str(ds.pathobj.parent / "empty-remote"))
+    # Mimic a situation that can happen with an LFS remote. See gh-4199.
+    ds.repo.config.unset("remote.origin.fetch", where="local")
+    (ds.repo.pathobj / "foo").write_text("a")
+    ds.save()
+    ds.publish(to="origin")
```

Comment:
I wonder if it is worth to explicitly note that it is just a smoke test.  it could have been an explicit mock for .fetch verifying that it was not called

Reply:
> I wonder if it is worth to explicitly note that it is just a smoke test.

I don't find "smoke test" a useful description.  It's specific purpose it to make sure that the fetches do not fail in this situation.  If you revert the changes from this PR, you see the failure that the changes here are trying to prevent: the "fatal: Couldn't find remote ref HEAD" if no branch" error on master or the GitPython failure on maint.

> it could have been an explicit mock for .fetch verifying that it was not called

Ew